staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -116,6 +119,9 @@ type DiffOptions struct {
 	EnforceNamespace bool
 	Builder          *resource.Builder
 	Diff             *DiffProgram
+	Prune            bool
```

Comment:
That's a lot of things to track just for pruning, can we move `PruneWhiteList` inside `pruner`? And so that a `nil` `pruner` does nothing (to remove the boolean)

Reply:
I'm seeing below how these are actually command line flags, I'm curious if you have to use the same `options.Flag` mechanism for these if they don't fit well in these options, i.e. use `StringArray` rather than `StringArrayVar` maybe.

Reply:
Can we do the same thing you did for `WhiteList` to this `Prune` field?